js/apps/events-page/dataServices/registrationService.js

```diff
@@ -0,0 +1,28 @@
+((app) => {
+	'use strict';
+
+	const registrationService = (CONSTANTS, $http, $q) => {
+		const register = (registrationModel) => {
+			return $q((resolve, reject) => {
```

Comment:
Use $q like this, https://docs.angularjs.org/api/ng/service/$q#testing create a deferred variable and return the deferred promise.

Reply:
I'm skipping this for now, but I'll think about it and possibly address later. I don't see extra value in using `$q.defer()` over `$q((resolve, reject) => { ... }`, but I'm happy to discuss.

Reply:
@mxsnyder no problem, I think it's slightly cleaner (my preference).